lib/elixir/test/elixir/calendar_test.exs

```diff
@@ -98,6 +98,17 @@ defmodule NaiveDateTimeTest do
     assert NaiveDateTime.compare(ndt3, ndt1) == :gt
     assert NaiveDateTime.compare(ndt3, ndt2) == :gt
   end
+
+  test "to_iso8601/1" do
+    ndt1 = ~N[2000-04-16 12:34:15.1234]
+    ndt1 = put_in ndt1.calendar, FakeCalendar
+    assert_raise ArgumentError, """
```

Comment:
Heredoc `"""` will add extra newline at the end, we could write it like that:
```elixir
ndt = %{~N[2000-04-16 12:34:15.1234] | calendar: FakeCalendar}
message = "cannot convert #{inspect(ndt)} to the ISO 8601 format, because it does not use Calendar.ISO"
assert_raise ArgumentError, message, fn ->
   NaiveDateTime.to_iso8601(ndt)
end
```
WDYT?